internal/command/machine/leases.go

```diff
@@ -0,0 +1,201 @@
+		usage = "view"
+	)
+
+	cmd := command.New(usage, short, long, runLeaseView,
+		command.RequireSession,
+		command.LoadAppNameIfPresent,
+	)
+
+	// at least one arg is required but we can accept a list of machine ids
+	cmd.Args = cobra.MinimumNArgs(1)
+
+	flag.Add(
+		cmd,
+		flag.App(),
+		flag.AppConfig(),
+	)
+
+	return cmd
+}
+
+func newLeaseClear() *cobra.Command {
+	const (
+		short = "Clear machine leases"
+		long  = short + "\n"
+		usage = "clear"
+	)
+
+	cmd := command.New(usage, short, long, runLeaseClear,
+		command.RequireSession,
+		command.LoadAppNameIfPresent,
+	)
+
+	// at least one arg is required but we can accept a list of machine ids
+	cmd.Args = cobra.MinimumNArgs(1)
+
+	flag.Add(
+		cmd,
+		flag.App(),
+		flag.AppConfig(),
+	)
+
+	return cmd
+}
+
+func runLeaseView(ctx context.Context) (err error) {
+	var (
+		io      = iostreams.FromContext(ctx)
+		args    = flag.Args(ctx)
+		cfg     = config.FromContext(ctx)
+		appName = app.NameFromContext(ctx)
+		client  = client.FromContext(ctx).API()
+	)
+
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		return fmt.Errorf("could not get app: %w", err)
+	}
+
+	ctx, err = apps.BuildContext(ctx, app)
+	if err != nil {
+		return
+	}
+
+	flapsClient := flaps.FromContext(ctx)
+
+	var machines []*api.Machine
+	// Resolve machines
+	for _, machineID := range args {
+		machine, err := flapsClient.Get(ctx, machineID)
+		if err != nil {
+			return fmt.Errorf("could not get machine %s: %w", machineID, err)
+		}
+		machines = append(machines, machine)
+	}
+
+	var leases = make(map[string]*api.MachineLease)
+
+	for _, machine := range machines {
+		lease, err := flapsClient.FindLease(ctx, machine.ID)
+		if err != nil {
+			if strings.Contains(err.Error(), " lease not found") {
+				continue
+			}
+			return err
+		}
+		if lease == nil {
+			continue
+		}
+
+		leases[machine.ID] = lease
+	}
+
+	if cfg.JSONOutput {
+		return render.JSON(io.Out, leases)
+	}
+
+	if len(leases) == 0 {
+		fmt.Fprintln(io.Out, "No leases found")
+		return nil
+	}
+
+	rows := [][]string{}
+
+	for machine, lease := range leases {
+		expires := time.Unix(lease.Data.ExpiresAt, 0).Format(time.RFC3339)
+
+		rows = append(rows, []string{
+			machine,
+			lease.Data.Nonce,
+			lease.Data.Owner,
+			lease.Status,
+			expires,
+		})
+	}
+
+	_ = render.Table(io.Out, "", rows, "Machine", "Nonce", "Status", "Owner", "Expires")
+
+	return
+}
+
+func runLeaseClear(ctx context.Context) (err error) {
+	var (
+		io      = iostreams.FromContext(ctx)
+		args    = flag.Args(ctx)
+		appName = app.NameFromContext(ctx)
+		client  = client.FromContext(ctx).API()
+	)
+
+	app, err := client.GetAppCompact(ctx, appName)
+	if err != nil {
+		return fmt.Errorf("could not get app: %w", err)
+	}
+
+	ctx, err = apps.BuildContext(ctx, app)
+	if err != nil {
+		return
+	}
+
+	flapsClient := flaps.FromContext(ctx)
+
+	for _, machineID := range args {
+		lease, err := flapsClient.FindLease(ctx, machineID)
+		if err != nil {
```

Comment:
We shouldn't fail in the event err == "lease not found".

Reply:
I thought it should have the same UX as `fly machine list`.

Reply:
Seems like we'd want the same logic you have above:

```
	if strings.Contains(err.Error(), " lease not found") {
		continue
	}
	return err
```

Since there may be cases where some Machines have leases and some don't.

Reply:
My bad, I misread this.